borg/cache.py

```diff
@@ -390,6 +400,39 @@ def legacy_cleanup():
             self.do_cache = os.path.isdir(archive_path)
             self.chunks = create_master_idx(self.chunks)
 
+    def check_cache_compatibility(self):
+        my_features = Manifest.SUPPORTED_REPO_FEATURES
+        if self.ignored_features & my_features:
+            return False
```

Comment:
So per https://github.com/borgbackup/borg/pull/2134#issuecomment-303220459 

> So ignored_features & my_features would be the set of features that were not supported by the previous client. But why wipe the cache in this instance?

I.e. the repository has had a feature enabled, which the client just now learned, so as per ^ the cache might need rebuilding. But as per ^^ this can be made a case-to-case decision anyway. Ok. Now I get it :)